compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -135,9 +134,36 @@ class ExpressionCodegen(
             }
         }
         writeLocalVariablesInTable(info)
+        createLocalVariablesForParameters(startLabel)
         mv.visitEnd()
     }
 
+    private fun createLocalVariablesForParameters(startLabel: Label) {
+        var register = 0
+        if (!irFunction.isStatic) {
+            mv.visitLocalVariable("this", classCodegen.type.descriptor, null, startLabel, markNewLabel(), register++)
+        }
+        val extensionReceiverParameter = irFunction.extensionReceiverParameter
+        if (extensionReceiverParameter != null) {
+            register = visitParameter(extensionReceiverParameter, startLabel, register)
+        }
+        for (param in irFunction.valueParameters) {
+            register = visitParameter(param, startLabel, register)
+        }
+    }
+
+    private fun visitParameter(param: IrValueParameter, startLabel: Label, register: Int): Int {
+        val descriptor = param.descriptor
+        val joinedParameterNames = if (descriptor is ValueParameterDescriptor)
+            ValueParameterDescriptorImpl.joinParameterNamesOrNull(descriptor) else null
+        val type = typeMapper.mapType(descriptor)
+        mv.visitLocalVariable(
+            if (joinedParameterNames == null) param.name.asString() else "\$$joinedParameterNames",
+            type.descriptor, null, startLabel, markNewLabel(), register
```

Comment:
It's should be possible to find register from 'frame', see 'findLocalIndex' (maybe except 'this' parameter)

Reply:
done